File: src/Java/Behavioral/Strategy/BatchMessageSender.java
package Java.Behavioral.Strategy;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class BatchMessageSender {

    private List<Message> messageList;

    public BatchMessageSender() {
        messageList = new ArrayList<>();
    }

    public void addMessage(Message message) {
        messageList.add(message);
    }

    public void addMessage(List<Message> messages) {
        messageList.addAll(messages);
    }

    public void sendBatch() {
        Calendar calendar = Calendar.getInstance();

        for (Message m : messageList) {
            m.setDateTimeSent(new SimpleDateFormat("dd/MM/yyyy HH:mm:ss").format(calendar.getTime()));
            m.getDispatchStrategy().send(m);
        }
    }
}

File: src/Kotlin/strategyPattern/BatchMessageSender.kt
package Kotlin.strategyPattern

import java.text.SimpleDateFormat
import java.util.*

class BatchMessageSender {

    var messageList: ArrayList<Message> = ArrayList()

    fun addMessage(message: Message) {
        messageList.add(message)
    }

    fun addMessage(messages: ArrayList<Message>) {
        messageList.addAll(messages)
    }

    fun sendBatch() {
        val calendar = Calendar.getInstance()

        for (m: Message in messageList) {
            val outputFmt = SimpleDateFormat("dd/MM/yyyy HH:mm:ss")
            val timeString = outputFmt.format(calendar.time)
            m.dateTimeSent = timeString
            m.dispatchStrategy.send(m)
        }
    }
}